xilem/examples/variable_clock.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This example uses variable fonts in a touch sensitive digital clock.
+
+use masonry::parley::{
+    fontique::Weight,
+    style::{FontFamily, FontStack},
+};
+use time::{error::IndeterminateOffset, macros::format_description, OffsetDateTime, UtcOffset};
+use winit::error::EventLoopError;
+use xilem::{
+    view::{button, flex, label, prose, sized_box, variable_label, Axis, FlexExt, FlexSpacer},
+    Color, EventLoop, EventLoopBuilder, WidgetView, Xilem,
+};
+
+/// The state of the application, owned by Xilem and updated by the callbacks below.
+struct Clocks {
+    /// The font [weight](Weight) used for the values.
+    weight: f32,
+    /// The current UTC offset on this machine.
+    local_offset: Result<UtcOffset, IndeterminateOffset>,
+    /// The current time.
+    now_utc: OffsetDateTime,
+}
+
+impl Clocks {
+    fn local_now(&self) -> OffsetDateTime {
+        match self.local_offset {
+            Ok(offset) => self.now_utc.to_offset(offset),
+            Err(_) => self.now_utc,
+        }
+    }
+}
+
+/// A possible timezone, with an offset from UTC.
+struct TimeZone {
+    /// An approximate region which this offset applies to.
+    region: &'static str,
+    /// The offset from UTC
+    offset: time::UtcOffset,
+}
+
+impl TimeZone {
+    fn view(&self, data: &mut Clocks) -> impl WidgetView<Clocks> {
+        let date_time_in_self = data.now_utc.to_offset(self.offset);
```

Comment:
Should this be updated every second?

Reply:
Indeed. This is planned, you just reviewed before I got around to it :)